name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/DegiroPDFExtractor.java

```diff
@@ -677,58 +677,322 @@ private void addPortfolioTransactions()
+                                            + "[\\w]{3} [\\w]{4} "
+                                            + "(?<shares>-?[.\\d]+[.\\d]*) "
+                                            + "-?([\\d]+,[\\d]{3}.\\d+|\\d+.\\d+) "
+                                            + "[\\w]{3}.* "
+                                            + "-?(?<amountFx>([\\d]+,[\\d]{3}.\\d+|\\d+.\\d+)) "
+                                            + "(?<currency>[\\w]{3}).* "
+                                            + "-?([\\d]+,[\\d]{3}.\\d+|\\d+.\\d+) "
+                                            + "[\\w]{3} "
+                                            + "(?<exchangeRate>[.\\d]+[.\\d]*) "
+                                            + "-?(?<amount>([\\d]+,[\\d]{3}.\\d+|\\d+.\\d+)) "
+                                            + "(?<currencyAccount>[\\w]{3})$")
+                            .assign((t, v) -> {
+                                t.setSecurity(getOrCreateSecurity(v));
+                                t.setDate(asDate(v.get("date")));
+                                if (v.get("shares").startsWith("-"))   
+                                {
+                                    t.setType(PortfolioTransaction.Type.SELL);
+                                    t.setShares(asShares(v.get("shares").replaceFirst("-", "")));   
+                                }
+                                else 
+                                {
+                                    t.setShares(asShares(v.get("shares"))); 
+                                }
+                                t.setCurrencyCode(asCurrencyCode(v.get("currencyAccount"))); 
+                                t.setAmount(asAmount(ConvertAmount(v.get("amount"))));
+
+                                long amountFx = asAmount(ConvertAmount(v.get("amountFx"))); 
+                                String currencyFx = asCurrencyCode(v.get("currency"));
+
+                                if (currencyFx.equals(t.getPortfolioTransaction().getSecurity().getCurrencyCode()))
+                                {
+                                    Money amount = Money.of(asCurrencyCode(v.get("currencyAccount")), asAmount(ConvertAmount(v.get("amount"))));  
+                                    BigDecimal exchangeRate = BigDecimal.ONE.divide(asExchangeRate(ConvertExchangeRate(v.get("exchangeRate"))), 10, RoundingMode.HALF_DOWN); 
+                                    Money forex = Money.of(asCurrencyCode(v.get("currency")), amountFx);
+                                    Unit grossValue = new Unit(Unit.Type.GROSS_VALUE, amount, forex, exchangeRate);
+                                    t.getPortfolioTransaction().addUnit(grossValue);
+                                }
                             })
                         )
 
                         .wrap(t -> new BuySellEntryItem(t)));
     }
+
+    @SuppressWarnings("nls")
+    private String ConvertAmount(String inputAmount)
```

Comment:
Java convention is: a method should start with the lower case letter, possibly a verb that does something.
I will just merge it now (as I am only in the web viewer) but it would be nice if you can send a minor PR to fix this. thanks.